cardano-cli/src/Cardano/CLI/EraBased/Common/Option.hs

```diff
@@ -1740,21 +1740,30 @@ pOperationalCertificateFile :: Parser (File () direction)
 pOperationalCertificateFile =
   File <$> parseFilePath "op-cert-file" "Filepath of the node's operational certificate."
 
-pKeyOutputFormat :: Parser KeyOutputFormat
+pKeyOutputFormat :: Parser (Vary [FormatBech32, FormatTextEnvelope])
 pKeyOutputFormat =
-  Opt.option readKeyOutputFormat $
-    mconcat
-      [ Opt.long "key-output-format"
-      , Opt.metavar "STRING"
-      , Opt.help $
-          mconcat
-            [ "Optional key output format. Accepted output formats are \"text-envelope\" "
-            , "and \"bech32\" (default is \"text-envelope\")."
-            ]
-      , Opt.value KeyOutputFormatTextEnvelope
-      ]
+  pFormatFlagsExt
+    "key output"
+    p
+    [ flagKeyOutputBech32
+    , flagKeyOutputTextEnvelope & setDefault
+    ]
+ where
+  p =
+    Opt.option
+      deprecatedReadKeyOutputFormat
+      $ mconcat
+        [ Opt.long "key-output-format"
+        , Opt.metavar "STRING"
+        , Opt.help $
+            mconcat
+              [ "Optional key output format. Accepted output formats are \"text-envelope\" "
+              , "and \"bech32\" (deprecated)."
```

Comment:
Why bech32 is deprecated?

Reply:
`bech32` isn't deprecated.  The `--key-output-format` switch is.

Reply:
Ah, the phrasing was slightly confusing. Maybe it should say "deprecated flag" or something.

Reply:
Fixed up the message to be more clear.